src/dune_rules/merlin.ml

```diff
@@ -122,12 +134,22 @@ let make ?(requires = Ok []) ~flags ?(preprocess = Preprocess.No_preprocessing)
         flags
       |> Ocaml_flags.common
   in
+  let extensions =
+    Dialect.DB.fold dialects ~init:Extensions.Set.empty ~f:(fun d s ->
+        let impl = Dialect.extension d Ml_kind.Impl in
+        let intf = Dialect.extension d Ml_kind.Intf in
+        if impl = ".ml" && intf = ".mli" then
```

Comment:
We can use `Dialect.extension Dialect.ocaml Impl` instead of hardcoding `".ml"`, etc.